lib/bolt/config/transport/base.rb

```diff
@@ -116,24 +119,21 @@ def self.options
           assert_type
         end
 
-        # Validates that each option is the correct type. Types are loaded from the OPTIONS hash.
+        # Validates that each option is the correct type. Types are loaded from the TRANSPORT_OPTIONS hash.
         private def assert_type
           @config.each_pair do |opt, val|
-            next unless (type = self.class.options.dig(opt, :type))
-
-            # Options that accept a Boolean value are indicated by the type TrueClass, so we
-            # need some special handling here to check if the value is either true or false.
-            if type == TrueClass
-              unless [true, false].include?(val)
-                raise Bolt::ValidationError,
-                      "#{opt} must be a Boolean true or false, received #{val.class} #{val.inspect}"
-              end
-            else
-              unless val.nil? || val.is_a?(type)
-                raise Bolt::ValidationError,
-                      "#{opt} must be a #{type}, received #{val.class} #{val.inspect}"
-              end
+            types = Array(TRANSPORT_OPTIONS.dig(opt, :type)).compact
+
+            next if val.nil? || types.empty? || types.include?(val.class)
```

Comment:
It seems weird that we don't err when a value is nil? I see that's the same as the old behavior but I'm not sure what a valid use case is for having a config value be nil...

Reply:
Ya, I really regret not commenting that when I wrote the class... there's a very particular edge case that it covers, but I can't remember what it was.